File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/Immunization.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Immunization : DomainResource, IResource
{
    public required Base.ResourceReference Patient { get; set; }
    public bool? IsSubpotent { get; set; }
    public Base.CodeableConcept? ReportOrigin { get; set; }
    public Immunization_ProtocolApplied[]? ProtocolApplied { get; set; }
    public Base.CodeableConcept? Site { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public required Base.CodeableConcept VaccineCode { get; set; }
    public Base.Quantity? DoseQuantity { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public Base.CodeableConcept? StatusReason { get; set; }
    public Base.CodeableConcept? Route { get; set; }
    public string? Recorded { get; set; }
    public Base.CodeableConcept[]? ProgramEligibility { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public bool? PrimarySource { get; set; }
    public required string Status { get; set; }
    public string? LotNumber { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? Manufacturer { get; set; }
    public Immunization_Education[]? Education { get; set; }
    public string? OccurrenceString { get; set; }
    public Immunization_Reaction[]? Reaction { get; set; }
    public Base.ResourceReference? Location { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public Base.CodeableConcept? FundingSource { get; set; }
    public Base.CodeableConcept[]? SubpotentReason { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrenceString is not null)
            {
                return OccurrenceString;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                OccurrenceString = (string)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public string? ExpirationDate { get; set; }
    public Immunization_Performer[]? Performer { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class Immunization_ProtocolApplied : BackboneElement
    {
        public string? SeriesDosesPositiveInt { get; set; }
        public string? DoseNumberPositiveInt { get; set; }
        public string? Series { get; set; }
        public object? DoseNumber    
        {
            get
            {
                if (DoseNumberPositiveInt is not null)
                {
                    return DoseNumberPositiveInt;
                }
        
                if (DoseNumberString is not null)
                {
                    return DoseNumberString;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    DoseNumberPositiveInt = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    DoseNumberString = (string)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.ResourceReference? Authority { get; set; }
        public string? DoseNumberString { get; set; }
        public string? SeriesDosesString { get; set; }
        public object? SeriesDoses    
        {
            get
            {
                if (SeriesDosesPositiveInt is not null)
                {
                    return SeriesDosesPositiveInt;
                }
        
                if (SeriesDosesString is not null)
                {
                    return SeriesDosesString;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    SeriesDosesPositiveInt = (string)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    SeriesDosesString = (string)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.CodeableConcept[]? TargetDisease { get; set; }
    }

    public class Immunization_Education : BackboneElement
    {
        public string? Reference { get; set; }
        public string? DocumentType { get; set; }
        public string? PublicationDate { get; set; }
        public string? PresentationDate { get; set; }
    }

    public class Immunization_Reaction : BackboneElement
    {
        public string? Date { get; set; }
        public Base.ResourceReference? Detail { get; set; }
        public bool? Reported { get; set; }
    }

    public class Immunization_Performer : BackboneElement
    {
        public required Base.ResourceReference Actor { get; set; }
        public Base.CodeableConcept? Function { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Immunization.py
from typing import Optional, List
from base import Annotation
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class Immunization_ProtocolApplied(BackboneElement):
    series_doses_positive_int: Optional[int] = None
    dose_number_positive_int: Optional[int] = None
    series: Optional[str] = None
    authority: Optional[Reference] = None
    dose_number_string: Optional[str] = None
    series_doses_string: Optional[str] = None
    target_disease: Optional[List[CodeableConcept]] = None

class Immunization_Education(BackboneElement):
    reference: Optional[str] = None
    document_type: Optional[str] = None
    publication_date: Optional[str] = None
    presentation_date: Optional[str] = None

class Immunization_Reaction(BackboneElement):
    date: Optional[str] = None
    detail: Optional[Reference] = None
    reported: Optional[bool] = None

class Immunization_Performer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None

class Immunization(DomainResource):
    patient: Reference
    is_subpotent: Optional[bool] = None
    report_origin: Optional[CodeableConcept] = None
    protocol_applied: Optional[List[Immunization_ProtocolApplied]] = None
    site: Optional[CodeableConcept] = None
    encounter: Optional[Reference] = None
    vaccine_code: CodeableConcept
    dose_quantity: Optional[Quantity] = None
    reason_code: Optional[List[CodeableConcept]] = None
    status_reason: Optional[CodeableConcept] = None
    route: Optional[CodeableConcept] = None
    recorded: Optional[str] = None
    program_eligibility: Optional[List[CodeableConcept]] = None
    note: Optional[List[Annotation]] = None
    primary_source: Optional[bool] = None
    status: str
    lot_number: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    manufacturer: Optional[Reference] = None
    education: Optional[List[Immunization_Education]] = None
    occurrence_string: Optional[str] = None
    reaction: Optional[List[Immunization_Reaction]] = None
    location: Optional[Reference] = None
    occurrence_date_time: Optional[str] = None
    funding_source: Optional[CodeableConcept] = None
    subpotent_reason: Optional[List[CodeableConcept]] = None
    expiration_date: Optional[str] = None
    performer: Optional[List[Immunization_Performer]] = None
    reason_reference: Optional[List[Reference]] = None